fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -233,9 +246,8 @@ public void writeData(long id, int block, @Nonnull byte[] value) {
             if (name.endsWith(".cfs") || name.endsWith(".si") || name.endsWith(".cfe")) {
                 try {
                     readBlock(name, CompletableFuture.completedFuture(fileReference), 0);
-                } catch (IOException ioe) {
-                    LOGGER.warn("Cannot Prefetch resource={}", name);
-                    LOGGER.warn("Prefetch Error", ioe);
+                } catch (RecordCoreException e) {
+                    LOGGER.warn(KeyValueLogMessage.of("Exception thrown during prefetch", "resource", name, "exception", e));
```

Comment:
I believe the exception is supposed to go outside `KeyValueLogMessage`
```suggestion
                    LOGGER.warn(KeyValueLogMessage.of("Exception thrown during prefetch", "resource", name), e);
```